scripts/diff-workflow-against-master.sh

```diff
@@ -0,0 +1,471 @@
+#         gcc/clang/vistual-studio warnings and errors, and diff them.
+#
+function main() {
+	# Setup all pre-requisites
+	init
+	echo "Comparing branch $local_branch with master"
+	echo ""
+
+	# Fetch the workflows, to be used throughout the run
+	fetch_workflows
+
+	# Step through each workflow
+	for workflow_id in "${!workflows[@]}"; do
+		workflow_name="${workflows[$workflow_id]}"
+		echo -e "${bold}${workflow_name}${reset} workflow [$workflow_id]"
+
+		# Within the workflows, we're interested in finding the newest subset of
+		# runs that match our current branch as well as the master branch.
+		for branch in master current; do
+			if [[ "$branch" == "current" ]]; then
+				branch_name="$local_branch"
+			else
+				branch_name="master"
+			fi
+
+			# Get the run identifier for the given workflow and branch
+			fetch_workflow_run "$workflow_id" "$branch_name"
+			if [[ -z "${run_id:-}" ]]; then
+				echo "  \`- no runs found for $branch_name"
+				continue
+			fi
+			[[ "$branch" == "master" ]] && joiner="|-" || joiner="\`-"
+			echo "  $joiner found latest $branch_name run [$run_id]"
+
+			# Download the artifacts produced during the selected run
+			fetch_run_artifacts
+			[[ "$branch" == "master" ]] && joiner="|" || joiner=" "
+			for artifact_name in "${!artifacts[@]}"; do
+				artifact_url="${artifacts[$artifact_name]}"
+				asset_file="$assets_dir/$workflow_name-$artifact_name-$branch.zip"
+				download "$artifact_url" "$asset_file"
+				unpack "$asset_file"
+				echo -e "  $joiner     - unpacking ${cyan}${artifact_name}${reset} asset"
+			done
+
+			# Download the logs for the jobs within the selected run
+			for conclusion in failure success; do
+				if ! fetch_run_jobs "$conclusion"; then
+					echo "      \`- skipped $job_id $conclusion"
+					continue
+				fi
+				[[ "$conclusion" == "success" ]] && color="${green}" || color="${red}"
+				for job_id in "${!jobs_array[@]}"; do
+					job_name="${jobs_array[$job_id]}"
+					echo -e "  $joiner     - fetching  ${color}${job_name}${reset} ${conclusion} log"
+					log_file="$logs_dir/$workflow_name-$job_name-$branch-$conclusion.log"
+					successful_master_log="$logs_dir/$workflow_name-$job_name-master-success.log"
+					fetch_job_log "$job_id" "$log_file"
+
+					# In the event we've found a failed job, try to diff it against a prior
+					# successful master job of the equivalent workflow and job-type.
+					if [[ "$conclusion" == "failure"
+					&& -f "$log_file"
+					&& -f "$successful_master_log" ]]; then
+						sanitized_branch_name="$(sanitize_name "$branch_name")"
+						diff_file="$diffs_dir/$workflow_name-$job_name-$sanitized_branch_name-vs-master.log"
+						diff "$log_file" "$successful_master_log" > "$diff_file" || true
+						echo -e "        - diffed    ${yellow}$diff_file${reset}"
+					fi
+				done # jobs_array
+			done # conclusion types
+			echo "  $joiner"
+		done # branch types
+	done # workflows
+}
+
+main
```

Comment:
Use `main "$@"`; we don't handle any parameters yet, but let's prepare for it.

Reply:
Makes sense; will do.